browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -91,6 +91,14 @@ void AIChatUIPageHandler::HandleVoiceRecognition(
 #endif
 }
 
+void AIChatUIPageHandler::ConversationExists(
+    const std::string& conversation_uuid,
+    ConversationExistsCallback callback) {
+  std::move(callback).Run(
+      AIChatServiceFactory::GetForBrowserContext(profile_)->GetConversation(
```

Comment:
This is only checking if a ConversationHandler is in memory for the conversation. Need to use the async version of `GetConversation` or, perhaps better, add `ConversationExists` to `Service` and call it directly

Reply:
The slight issue with that is that it will cause the whole conversation to be loaded from database. Since we know this is called just before the conversation is shown then that's likely ok (or maybe even a positive optimization?), but in an isolated purpose of the function it's negative.

Reply:
It will only cause it to be loaded from the database if it isn't already loaded - as an added optimization on the frontend, I check whether the conversation is in the list of visible conversations before doing the mojom roundtrip for checking existence.

For example, in the only place this function is currently used, it will not result in a request to the database, as the conversation will already be in memory (which is why it was working without me using the async method).

I think I'd prefer to leave this as is for now, rather than adding a new method to the database for checking whether a conversation exists. When you think about it, most of the time you check when something exists you're probably going to do something with it afterwards.

If this does become a problem (i.e. we start calling this in a tight loop), lets optimize it then. The optimizations will also be different depending on how we end up using it (if its the single call that's the problem, we can add a db query, as you suggest - if we're making lots of calls maybe we need to cache a set of the conversation ids).

Reply:
Oh I see so it's because the "visible conversations" list hasn't loaded yet?

Reply:
Nah, its because the new conversation won't be in the list of visible conversations because it has no content (which means we'd normally redirect to the `/` url as this looks like its just a new conversation).